UM/View/RenderBatch.py

```diff
@@ -302,21 +261,79 @@ def _renderItem(self, item: Dict[str, Any]):
                 Logger.log("e", "Attribute with name [%s] uses non implemented type [%s]." % (attribute["opengl_name"], attribute["opengl_type"]))
                 self._shader.disableAttribute(attribute["opengl_name"])
 
-        if mesh.hasIndices():
-            if self._render_range is None:
-                if self._render_mode == self.RenderMode.Triangles:
-                    self._gl.glDrawElements(self._render_mode, mesh.getFaceCount() * 3, self._gl.GL_UNSIGNED_INT, None)
-                else:
-                    self._gl.glDrawElements(self._render_mode, mesh.getFaceCount(), self._gl.GL_UNSIGNED_INT, None)
-            else:
-                if self._render_mode == self.RenderMode.Triangles:
-                    self._gl.glDrawRangeElements(self._render_mode, self._render_range[0], self._render_range[1], self._render_range[1] - self._render_range[0], self._gl.GL_UNSIGNED_INT, None)
-                else:
-                    self._gl.glDrawElements(self._render_mode, self._render_range[1] - self._render_range[0], self._gl.GL_UNSIGNED_INT, None)
-        else:
-            self._gl.glDrawArrays(self._render_mode, 0, vertex_count)
+    def _vertexBuffersSetup(self, mesh: Any) -> Optional[QOpenGLVertexArrayObject]:
```

Comment:
`Any` is a bit lazy here, when you obviously expect `MeshData`

Reply:
Oops! Fixed in https://github.com/Ultimaker/Uranium/pull/714/commits/8611188f88ec11dbef37bcd40dd96f79d17540cd

There's a joke about lazy evaluation in there ... _but I just couldn't be bothered._